src/tools/rbd_mirror/image_sync/ImageCopyRequest.cc

```diff
@@ -146,7 +149,22 @@ void ImageCopyRequest<I>::send_object_copies() {
       }
     }
     complete = (m_current_ops == 0);
+
+    if (!complete) {
+      m_update_sync_ctx = new FunctionContext([this](int r) {
+          this->send_update_sync_point();
+      });
+    }
   }
+
+  {
+    Mutex::Locker timer_locker(*m_timer_lock);
+    if (m_update_sync_ctx) {
```

Comment:
I think this is going to cause helgrind warning since you sometimes hold only ```m_timer_lock``` when touching ```m_update_sync_ctx```, sometimes you hold ```m_lock```, and sometimes you hold both.